enforcer.go

```diff
@@ -271,38 +287,39 @@ func (e *Enforcer) ClearPolicy() {
 
 // LoadPolicy reloads the policy from file/database.
 func (e *Enforcer) LoadPolicy() error {
-	oldModel := e.model
-	e.model = model.NewModel()
-	e.model.SetLogger(oldModel.GetLogger())
+	newModel := model.NewModel()
+	e.model.Copy(&newModel)
+	newModel.ClearPolicy()
 
 	var err error
+	var rmMap RoleManagerMap
 	defer func() {
-		if err != nil {
-			e.model = oldModel
+		if err == nil {
+			e.model = newModel
+			e.rmMap = rmMap
 		}
 	}()
 
-	if err = e.model.LoadModelFromText(oldModel.ToText()); err != nil {
-		return err
-	}
-	if err = e.adapter.LoadPolicy(e.model); err != nil && err.Error() != "invalid file path, file path cannot be empty" {
+	if err = e.adapter.LoadPolicy(newModel); err != nil {
 		return err
 	}
 
-	if err = e.model.SortPoliciesByPriority(); err != nil {
+	if err = newModel.SortPoliciesByPriority(); err != nil {
 		return err
 	}
 
-	if err = e.clearRmMap(); err != nil {
-		return err
+	rmMap = make(map[string]rbac.RoleManager)
+	for ptype := range newModel["g"] {
```

Comment:
Why not use this to init rmMap?

https://github.com/casbin/casbin/blob/beedee2c0f86039b927863a4e9931e91421456a3/enforcer.go#L379-L387

Plz avoid duplicated code.

Reply:
Because we need to create a new rmMap instead of e.rmMap, so I add this code, let me optimize that.